Add Supporter Wall section to TwG edit screen

nfmohit opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #5736 

## Relevant technical choices

This PR adds the supporter wall widget section to the Thank with Google module edit screen.

Besides following the IB, this PR also checks if the `getSupporterWallSidebars` selector is undefined (resolving) and displays a `small` `<ProgressBar />` in that case.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

tofumatt reviewed (changes requested):
Looks good, just a few small questions/tweaks and then this should be good-to-go 👍🏻

nfmohit commented:
> Looks good, just a few small questions/tweaks and then this should be good-to-go 👍🏻

I have updated the PR and addressed the suggestions. Thank you, @tofumatt!

tofumatt reviewed (approved):
Awesome, looks great, thanks! 👍🏻
